### PR TITLE
Allow UUID to be used as id parameter

### PR DESCRIPTION
**Pull request by patrickli**
By adding dash as allowed character in the regex.

### CODE REVIEW

**Comment by sergiostrampelli** [comment]
see #2697

**Comment by rande** [comment]
@patrickli can you add unit test ?

**Comment by patrickli** [comment]
Closing in favour of #2697

**Comment by patrickli** [comment]
Reopen in favour of #2697.

File: Admin/Admin.php [hunk 1]
```diff
@@ -1742,11 +1742,11 @@ public function setSubject($subject)
     public function getSubject()
     {
         if ($this->subject === null && $this->request) {
-            $id = $this->request->get($this->getIdParameter());
-            if (!preg_match('#^[0-9A-Fa-f]+$#', $id)) {
+            if (!$this->subject = $this->getModelManager()->find(
```
Comment (pulzarraider) [inline Admin/Admin.php]: This is not very optimal if there is no id parameter in the request. Some check for empty id should be before calling the `find` method.

Comment (patrickli) [inline Admin/Admin.php]: Thanks for the tip. Updated.

**Comment by rande** [comment]
duplicate https://github.com/sonata-project/SonataAdminBundle/commit/8e7832ff7bb109a6eca7629196b398f21866a9cb

**Pull request closed without merging**